File: ImperatorToCK3/Source/Mappers/NicknameMapper/NicknameMapping.cpp
#include "NicknameMapping.h"
#include "ParserHelpers.h"
#include "CommonRegexes.h"



mappers::NicknameMapping::NicknameMapping(std::istream& theStream) {
	registerKeys();
	parseStream(theStream);
	clearRegisteredKeywords();
}


void mappers::NicknameMapping::registerKeys() {
	registerKeyword("ck3", [this](std::istream& theStream) {
		ck3Nickname = commonItems::getString(theStream);
	});
	registerKeyword("imp", [this](std::istream& theStream) {
		impNicknames.insert(commonItems::getString(theStream));
	});
	registerMatcher(commonItems::catchallRegexMatch, commonItems::ignoreItem);
}

File: ImperatorToCK3/Mappers/Nickname/NicknameMapping.cs
using commonItems;
using System.Collections.Generic;

namespace ImperatorToCK3.Mappers.Nickname;

public sealed class NicknameMapping {
	public SortedSet<string> ImperatorNicknames { get; } = new();
	public string? CK3Nickname { get; private set; }

	public NicknameMapping(BufferedReader mappingReader) {
		var parser = new Parser();
		parser.RegisterKeyword("ck3", reader => CK3Nickname = reader.GetString());
		parser.RegisterKeyword("ir", reader => ImperatorNicknames.Add(reader.GetString()));
		parser.RegisterRegex(CommonRegexes.Catchall, ParserHelpers.IgnoreAndLogItem);
		parser.ParseStream(mappingReader);
	}
}